### PR TITLE
Contact ID column to plugin integration mapping

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/pull/5310
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:

This PR add the contact ID in plugin mapping. That can help to send mautic contact ID into a CRM for instance.  This PR comes in favor of https://github.com/mautic/mautic/pull/5310 (@npracht  please close old PR )

#### Steps to test this PR:

1. Apply the PR
2. Configure a plugin
3. Go to contact field mapping
4. Notice that the Mautic contact ID is available and sync to integration

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Feature Request

**Comment by kuzmany** [comment]
Label: Ready to test

**Comment by npracht** [comment]
@justinfortes it has been modified as requested by Alan in #5310. Could we add that as candidate for 2.13.0 ?
Thanks !

File: app/bundles/PluginBundle/Model/PluginModel.php [hunk 1]
```diff
@@ -62,7 +70,10 @@ public function getPermissionBase()
      */
     public function getLeadFields()
     {
-        return $this->leadFieldModel->getFieldList();
+        $leadFields         =  $this->leadFieldModel->getFieldList();
+        $leadFields['Core'] = ['id' => $this->translator->trans('mautic.lead.report.contact_id')] + $leadFields['Core'];
```
Comment (npracht) [inline app/bundles/PluginBundle/Model/PluginModel.php]: Hello @kuzmany sometimes i have an issue but it is not reproducable all the time. It throws a 500.

>[2018-03-12 11:13:47] mautic.ERROR: Symfony\Component\Debug\Exception\FatalThrowableError: Unsupported operand types - in file /home/path/app/bundles/PluginBundle/Model/PluginModel.php - at line 74 [] []

It seems that the `+` at that line generates the issue. But we couldn't reproduce all the time... I don't know what to do :)

Comment (npracht) [reply app/bundles/PluginBundle/Model/PluginModel.php]: We found something !

It is working in english but not in other languages. We're having a look on it and we comeback to you :)

Comment (npracht) [reply app/bundles/PluginBundle/Model/PluginModel.php]: @kuzmany if you change `line 74` by the following code, it solves the issue :)
```php
$leadFields[$this->translator->trans('mautic.lead.field.group.core')]['id'] = $this->translator->trans('mautic.lead.report.contact_id');
```

Comment (kuzmany) [reply app/bundles/PluginBundle/Model/PluginModel.php]: Fixed

Comment (kuzmany) [reply app/bundles/PluginBundle/Model/PluginModel.php]: Fixed

**Comment by kuzmany** [comment]
@npracht I can't :(

**Comment by escopecz** [comment]
@kuzmany we cannot merge this if it's removing composer.lock. Try to either get rid of the commit that removes it or take the composer.lock from staging and put it back :)

There are also some CS Fixer issues. They are caused by new version of CS Fixer. So once you put composer.lock back, run `composer install`. Then make small edits like new lines to those 3 reported files and commit. It will fix those CS issues.

**Comment by kuzmany** [comment]
@escopecz but how I can delete composer.lock, it's not in my branch https://github.com/kuzmany/mautic/tree/plugin-id-mapping . I tried create empty file, commit, then delete it, but still exist :(

**Comment by escopecz** [comment]
Why would you want to delete it? It should be there. So add it back from the staging, please :)

**Comment by npracht** [comment]
👏

**Comment by kuzmany** [comment]
@escopecz ready to test ;)

**Comment by alanhartless** [comment]
Does this lock out being able to overwrite Mautic's ID? We just need to make sure that's not possible. Also, there's a way to disable the direction arrows as we do it for the mautic timeline link so should do that for UI sake as well.

**Comment by kuzmany** [comment]
Label: Pending feedback

**Comment by kuzmany** [comment]
Label: WIP

**Comment by kuzmany** [comment]
PR update with @alanhartless recommendation changes https://github.com/mautic/mautic/pull/5310/files#r156728497
Please re-test

**Comment by escopecz** [comment]
@kuzmany there is some Travis check failing

**Comment by kuzmany** [comment]
@escopecz fixed

**Comment by npracht** [comment]
@escopecz can we add it as candidate for 2.14? It is ready since 2 weeks now :)

**Comment by npracht** [comment]
Label: Pending test confirmation

File: app/bundles/PluginBundle/Integration/AbstractIntegration.php [hunk 2]
```diff
@@ -2752,6 +2757,7 @@ public function getLeadDoNotContact($leadId, $channel = 'email')
     public function getCompoundMauticFields($lead)
     {
         if ($lead['internal_entity_id']) {
+            $lead['mauticContactId']                   = $lead['id'];
```
Comment (dongilbert) [inline app/bundles/PluginBundle/Integration/AbstractIntegration.php]: Should be $lead['internal_entity_id'] to remove question of ambiguity (just like the lines below). Getting the beta release out so moving this to 2.14.1

**Comment by heathdutton** [comment]
@kuzmany can you make that one requested change by @dongilbert? I'd really like to see this release w/ 2.14.1 and his request makes a lot of sense.

**Pull request merged**